src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NUM_ONE = 1;
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < DEFAULT_SIZE) {
+            elementData[size++] = value;
+        } else {
+            grow();
+            elementData[size++] = value;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (size < DEFAULT_SIZE) {
+            arrayCopyForAddByIndex(index);
+            elementData[index] = value;
+            size++;
+        } else {
+            grow();
+            arrayCopyForAddByIndex(index);
+            elementData[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (elementData.length > list.size()) {
+            for (int i = 0; i < list.size(); i++) {
+                elementData[size++] = list.get(i);
+            }
+        } else {
+            grow();
+            addAll(list);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        if (size > index) {
```

Comment:
you need to create second "check index" method (index >= size || index < 0). And remove this condition and exception from the methods. You can rename your first "chekIndex" method like "chekIndexForAdd".